### ISSUE TITLE
Question: Is this code base good for production usage?

### ISSUE THREAD

**Issue by ghost** [ExtraHop/rpcapd#1]
Thanks for sharing this project with the community. As I see, this projects hasn't been updated over last 4 years or so but was being used in ExtraHop products. Is this project going to be updated with latest changes if any? Is this  still being used in production at ExtraHop?

**Comment by daethnir** [comment]
Indeed, this is still used in production quite widely. We do have a number of improvements that we haven't pushed up here recently - I'll get that on the docket here in the next few weeks.

We'll happily accept suggestions/pull requests. What's your use case?

**Comment by ghost** [comment]
I am looking at grabbing packets and dumping them a time series db for offline analysis, an APM use case to figure out latency issues. From improvement point of view, the only feature I will probably add to rpcapd would be sampling at the source. 

Would love to know what are the improvements done by your team, happy to help in dev/testing of any features if they are still in the works.

**Comment by sunny137** [comment]
Couple more quick queries:

1. Does rpacpd support passive sniffing from a certain interface:port tuple? E.g. I have nginx running on my CentOS vm in AWS listening on port 80, is it possible to launch rpcapd as follows, I think this is not working in my setup for some reason. Some of the passive sniffing tools to work in this mode.

Example:
Netstat shows following,

tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      16870/nginx: master 

$sudo ./rpcapd -p 80 -n -a <remote_ip_address>,<remote_port>
Press CTRL + C to stop the server...
Connecting to host 104.197.110.40, port 80, using protocol Unspecified
bind(): Address already in use (code 98)
bind(): Address already in use (code 98)


2. I think rpcapd ships out packets over UDP, over a range of ports, correct? Is it possible to ship packets out over TCP as well?

**Comment by ghost** [comment]
Any inputs on the queries above would help me a lot! Please suggest.

**Comment by ghost** [comment]
Hello again, any updates on features in pipeline to be released here? Thank you.

**Comment by ghost** [comment]
We added end to end encryption to rpcapd after resolving some initial issues. Would love to know if we are re-inventing the wheel and you have already implemented these bits in your version which is being pushed up here.

**Comment by daluu** [comment]
Unfortunate that we haven't heard a response from @daethnir or ExtraHop here.

And unfortunate the OP's account is deleted as well. I wanted to ask some follow up questions regarding "we added end to end encryption to rpcapd after resolving some initial issues". Because rpcapd is now part of libpcap, and they have added TLS/SSL encryption support for rpcap (both rpcapd and rpcap clients). It would have been nice to check if the mentioned changes are similar or different from libpcap's changes and whether further improvements could be merged in there if the changes were permitted to be shared as open source.

For this repo, I would say aside from active mode here, without encryption and security improvements, there will always be security concerns for running this on production.
